File: lista de exercicios/questão 3/questao.go
package main

import "fmt"

type ContaBancaria struct {
    Titular string
    saldo   float64
}

func (c *ContaBancaria) Depositar(quantia float64) {
    c.saldo += quantia
}

func (c *ContaBancaria) Sacar(quantia float64) {
    if quantia <= c.saldo {
        c.saldo -= quantia
    } else {
        fmt.Println("Saldo insuficiente.")
    }
}

func (c ContaBancaria) ExibirSaldo() {
    fmt.Printf("Saldo atual: %.2f\n", c.saldo)
}

func main() {
    conta := ContaBancaria{Titular: "João", saldo: 500.0}
    conta.Depositar(200.0)
    conta.ExibirSaldo()
    conta.Sacar(100.0)
    conta.ExibirSaldo()
}

File: lista de exercicios/questão 3/questao.py
class ContaBancaria:
    def __init__(self, titular, saldo_inicial):
        self.__titular = titular
        self.__saldo = saldo_inicial

    def depositar(self, quantia):
        self.__saldo += quantia

    def sacar(self, quantia):
        if quantia <= self.__saldo:
            self.__saldo -= quantia
        else:
            print("Saldo insuficiente.")

    def exibir_saldo(self):
        print(f"Saldo atual: {self.__saldo}")

conta = ContaBancaria("João", 500.0)
conta.depositar(200.0)
conta.exibir_saldo()
conta.sacar(100.0)
conta.exibir_saldo()
